src/pages/RFOX/components/AddressSelection.tsx

```diff
@@ -197,16 +243,34 @@ export const AddressSelection: FC<AddressSelectionProps> = ({
         isDisabled={!maybeSelectedRuneAddress}
         value={maybeSelectedRuneAddress ?? ''}
       >
-        <AccountDropdown
-          defaultAccountId={maybeRuneAccountId}
-          assetId={thorchainAssetId}
-          onChange={handleAccountIdChange}
-          boxProps={boxProps}
-          buttonProps={buttonProps}
-        />
+        {(!Boolean(currentRuneAddress) && maybeRuneAccountId && setStepIndex) ||
+        accountNumber !== undefined ||
+        (!setStepIndex && runeAccounts.length) ? (
```

Comment:
If setStepIndex is not defined, we are not in the Stake scope, if we have some rune accounts we can still display the account selector

If not, display the custom address or "No address" copy!

This was the reason why metamask had a weird copy button without address, setStepIndex was not defined but we had no rune accounts if the snap is disabled